packages/fuels-abigen-macro/tests/harness.rs

```diff
@@ -1545,3 +1545,30 @@ async fn unit_type_enums() {
         .unwrap();
     assert_eq!(result.value, BimBamBoum::Boum());
 }
+
+#[tokio::test]
+// TODO(vnepveu): test for multiple assets
+async fn test_provider_balance_api() {
+    let mut wallet = LocalWallet::new_random(None);
+    let coins = setup_coins(wallet.address(), 21, 11);
+    let (provider, _) = setup_test_provider(coins.clone(), Config::local_node()).await;
+    wallet.set_provider(provider);
+    for (_utxo_id, coin) in coins {
+        let balance = wallet
+            .get_provider()
+            .unwrap()
+            .get_asset_balance(&wallet.address(), coin.asset_id)
+            .await;
```

Comment:
We can improve the ergonomics here. 

`wallet.get_provider().unwrap().get_asset_balance(&wallet.address(), asset_id)` is _way_ too much just to fetch the balance.

I see that `get_asset_balance()` is bound to `Provider`, which is good. But add a thin layer bound to `Wallet` so that (1) we can remove the need for `&wallet.address()` when using your own wallet and (2) we can remove the `get_provider()`. 

It would be just a `get_asset_balance(&self, asset_id: AssetId)` where self is `Wallet`. Then inside, it would call the provider's `get_asset_balance()` as-is. Calling `wallet.get_asset_balance(asset_id)` without having a provider set would throw an error.

Reply:
See comment above. To clarify I can maybe add a different worflow. There is an abstaction for the wallet obviously.

Reply:
Ok, so I'll write it "canonically", and add the unit tests later then!